### DISCUSSION TITLE
Bug: TypeError: Unexpected array.

### DISCUSSION THREAD

**Discussion by chenxsan** [eslint/eslint#16739] [Q&A] [answered]
### Environment

Node version: v18.12.1
npm version: 8.19.2
Local ESLint version: 8.30.0

### What parser are you using?

Default (Espree)

### What did you do?

I was trying to use predefined configurations from `typescript-eslint` according to the new Configuration file documentation here https://eslint.org/docs/latest/user-guide/configuring/configuration-files-new#using-predefined-configurations and https://eslint.org/blog/2022/08/new-config-system-part-2/#backwards-compatibility-utility:

```js
import typescriptEslintPlugin from '@typescript-eslint/eslint-plugin'
import { FlatCompat } from '@eslint/eslintrc'
import path from 'path'
import { fileURLToPath } from 'url'

// mimic CommonJS variables -- not needed if using CommonJS
const __filename = fileURLToPath(import.meta.url)
const __dirname = path.dirname(__filename)

const compat = new FlatCompat({
  baseDirectory: __dirname, // optional; default: process.cwd()
  resolvePluginsRelativeTo: __dirname, // optional
})

export default [
  compat.extends('plugin:@typescript-eslint/recommended')
]
```
Btw, you can reproduce it here https://stackblitz.com/edit/node-7ecwgq?file=eslint.config.js

Then I ran into this error:

```
$ npx eslint .
Oops! Something went wrong! :(

ESLint: 8.30.0

TypeError: Unexpected array.
```

because `compat.extends` returns an array here (there's a similar report here https://github.com/eslint/eslintrc/pull/99), and FlatESLint doesn't seem to support nested config array. 

And I noticed there're some inconsistency over different documentations:

1. [Configuration Files (New)](https://eslint.org/docs/latest/user-guide/configuring/configuration-files-new#configuration-file) mentions:

    > The ESLint configuration file is named eslint.config.js. It should be placed in the root directory of your project and export an array of [configuration objects](https://eslint.org/docs/latest/user-guide/configuring/configuration-files-new#configuration-objects).
 
    I.e., you can't use nested array of configuration objects.
2. [RFC - Config File Simplification](https://github.com/eslint/rfcs/tree/main/designs/2019-config-simplification#extending-another-config)

    > A config array is always flattened before being evaluated, so even though this example is a two-dimensional config array, it will be evaluated as if it were a one-dimensional config array.

    I.e., you can use nested config array.
3. [New config system part 2](https://eslint.org/blog/2022/08/new-config-system-part-2/#backwards-compatibility-utility) implies FlatESLint does support nested config array.

And I noticed [config-array](https://github.com/humanwhocodes/config-array#config-arrays) library has this `extraConfigTypes` which can be enabled to support nested config array, but it's not activated in this `eslint` repository.

### What did you expect to happen?

Nested config array to be supported as per RFC.

### What actually happened?

Nested config array is not supported at the moment.

### Participation

- [X] I am willing to submit a pull request for this issue.

### Additional comments

I'm not sure which one is true here, nested config array supported or not supported? If not, maybe we need to update the blog and eslintrc readme.

**Comment by nzakas** [comment, marked as answer]
Sorry about that, the documentation was actually incorrect about how to use the compat utility. You need to include `...` in front of each call to `compat.extends()`. The README and the blog post have now been updated:
https://github.com/eslint/eslintrc#usage
